frontend/src/metabase-lib/lib/Question.js

```diff
@@ -246,6 +246,25 @@ export default class Question {
     toUnderlyingData(): Question {
         return this.setDisplay("table");
     }
+
+    composeThisQuery(): ?Question {
```

Comment:
Of course using those magic constants is kinda un-zen but other than that I don't feel that there is anything fundamentally wrong in doing it this way. If you want, you could do some fanciness like
```
if (this.query() instanceof StructuredQuery && OtherConditionsSuggestedByCamsaul) {
    return this.query().useQuestionAsDataSource(this.id()).question()
} else {
    console.warn("Tried to create a composed question although this question cannot be nested")
    return null
}
```